### DISCUSSION TITLE
[Question] Recommended approach for invalidating lists after a change?

### DISCUSSION THREAD

**Discussion by tetchel** [n1ru4l/graphql-live-query#952] [Q&A] [answered]
Let's say I have an query `customers` which returns the list of customers, as an array of `Customer` GraphQLObjectTypes

Then I have the mutations `addCustomer`, `removeCustomer`, and `updateCustomer`.

After executing `addCustomer` or `removeCustomer`, I invalidate `Query.customers`, and the whole array is sent back over.
Is there an approach possible with this library which will allow just sending the new, added customer, or just sending a deletion message? 
Or do I have to use custom subscriptions for that?

Similarly, after executing `updateCustomer`, I can invalidate `Customer:${id}`. I see this also triggers sending the entire `customers` array, when I'd prefer to send just the changed element, and then replace the existing item in the `customers` query result cache locally.

The readme states,
>cache update code for applying the subscription result can eventually become pretty bloated (!!! especially, for adding and removing list items).

This library is definitely a way easier developer experience, but if it comes at the cost of sending way more data than necessary in these common cases, I may prefer using regular subscriptions.

**Comment by n1ru4l** [comment, marked as answer]
@tetchel This is solved via the https://github.com/n1ru4l/graphql-live-query/tree/main/packages/graphql-live-query-patch-jsondiffpatch middleware that ensures only patches are sent over the wire.

The trade-off here is that the server always stores the last execution result and diffs it against the new execution result. Hence there is more overhead going on on the server, but in contrast, the client is free of subscription update logic.
